sklearn/decomposition/_nmf.py

```diff
@@ -1729,3 +1774,616 @@ def inverse_transform(self, W):
+            If False, only W will be estimated, this corresponds to a call
+            to the 'transform' method.
+
+        Returns
+        -------
+        W : ndarray of shape (n_samples, n_components)
+            Transformed data.
+
+        H : ndarray of shape (n_components, n_features)
+            Factorization matrix, sometimes called 'dictionary'.
+
+        n_iter : int
+            Actual number of started iterations over the whole dataset.
+
+        n_steps : int
+            Number of mini-batches processed.
+        """
+        check_non_negative(X, "NMF (input X)")
+        self._check_params(X)
+
+        if X.min() == 0 and self._beta_loss <= 0:
+            raise ValueError(
+                "When beta_loss <= 0 and X contains zeros, "
+                "the solver may diverge. Please add small values "
+                "to X, or use a positive beta_loss."
+            )
+
+        n_samples, n_features = X.shape
+
+        # initialize or check W and H
+        W, H = self._check_w_h(X, W, H, update_H)
+        H_buffer = H.copy()
+
+        # Initialize auxiliary matrices
+        self._components_numerator = H.copy()
+        self._components_denominator = np.ones(H.shape, dtype=H.dtype)
+
+        # Attributes to monitor the convergence
+        self._ewa_cost = None
+        self._ewa_cost_min = None
+        self._no_improvement = 0
+
+        batches = gen_batches(n_samples, self._batch_size)
+        batches = itertools.cycle(batches)
+        n_steps_per_iter = int(np.ceil(n_samples / self._batch_size))
+        n_steps = self.max_iter * n_steps_per_iter
+
+        for i, batch in zip(range(n_steps), batches):
+
+            batch_cost = self._minibatch_step(X[batch], W[batch], H, update_H)
+
+            if update_H and self._minibatch_convergence(
+                X[batch], batch_cost, H, H_buffer, n_samples, i, n_steps
+            ):
+                break
+
+            H_buffer[:] = H
+
+        if self.fresh_restarts:
+            W = self._solve_W(X, H, self._transform_max_iter)
+
+        n_steps = i + 1
+        n_iter = int(np.ceil(n_steps / n_steps_per_iter))
+
+        return W, H, n_iter, n_steps
+
+    def transform(self, X):
+        """Transform the data X according to the fitted MiniBatchNMF model.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Data matrix to be transformed by the model.
+
+        Returns
+        -------
+        W : ndarray of shape (n_samples, n_components)
+            Transformed data.
+        """
+        check_is_fitted(self)
+        X = self._validate_data(
+            X, accept_sparse=("csr", "csc"), dtype=[np.float64, np.float32], reset=False
+        )
+
+        W = self._solve_W(X, self.components_, self._transform_max_iter)
+
+        return W
+
+    def partial_fit(self, X, y=None, W=None, H=None):
+        """Update the model using the data in X as a mini-batch.
```

Comment:
could we explain how a user could use `partial_fit` on a dataset to get the same result as running `fit` on its entirety? It would make it easier for people to decide how to use it.

Reply:
I improved the docstring and linked to the doc of incremental learning. Maybe we could add a section there to explain with more details how to use partial_fit, but I think it should be done in a separate PR since it concerns many estimators.